test/e2e_node/topology_manager_test.go

```diff
@@ -350,27 +501,252 @@ func runTopologyManagerSuiteTests(f *framework.Framework) {
+	}
+
+	deletePods(f, []string{pod.Name})
+}
+
+func isTopologyAffinityError(pod *v1.Pod) bool {
+	re := regexp.MustCompile(`Topology.*Affinity.*Error`)
+	return re.MatchString(pod.Status.Reason)
+}
+
+func getSRIOVDevicePluginConfigMap(cmFile string) *v1.ConfigMap {
+	cmData := testfiles.ReadOrDie(SRIOVDevicePluginCMYAML)
+	var err error
+
+	// the SRIOVDP configuration is hw-dependent, so we allow per-test-host customization.
+	framework.Logf("host-local SRIOV Device Plugin Config Map %q", cmFile)
+	if cmFile != "" {
+		cmData, err = ioutil.ReadFile(cmFile)
+		if err != nil {
+			framework.Failf("unable to load the SRIOV Device Plugin ConfigMap: %v", err)
+		}
+	} else {
+		framework.Logf("Using built-in SRIOV Device Plugin Config Map")
+	}
+
+	return readConfigMapV1OrDie(cmData)
+}
+
+func setupSRIOVConfigOrFail(f *framework.Framework, configMap *v1.ConfigMap) (*v1.Pod, string, int64) {
+	var err error
+
+	ginkgo.By(fmt.Sprintf("Creating configMap %v/%v", metav1.NamespaceSystem, configMap.Name))
+	if _, err = f.ClientSet.CoreV1().ConfigMaps(metav1.NamespaceSystem).Create(configMap); err != nil {
+		framework.Failf("unable to create test configMap %s: %v", configMap.Name, err)
+	}
+
+	serviceAccount := readServiceAccountV1OrDie(testfiles.ReadOrDie(SRIOVDevicePluginSAYAML))
+	ginkgo.By(fmt.Sprintf("Creating serviceAccount %v/%v", metav1.NamespaceSystem, serviceAccount.Name))
+	if _, err = f.ClientSet.CoreV1().ServiceAccounts(metav1.NamespaceSystem).Create(serviceAccount); err != nil {
+		framework.Failf("unable to create test serviceAccount %s: %v", serviceAccount.Name, err)
+	}
+
+	e2enode.WaitForNodeToBeReady(f.ClientSet, framework.TestContext.NodeName, 5*time.Minute)
+
+	dp := getSRIOVDevicePluginPod()
+	dp.Spec.NodeName = framework.TestContext.NodeName
+
+	ginkgo.By("Create SRIOV device plugin pod")
+	dpPod, err := f.ClientSet.CoreV1().Pods(metav1.NamespaceSystem).Create(dp)
+	framework.ExpectNoError(err)
+
+	sriovResourceName := ""
+	var sriovResourceAmount int64
+	ginkgo.By("Waiting for devices to become available on the local node")
+	gomega.Eventually(func() bool {
+		node := getLocalNode(f)
+		framework.Logf("Node status: %v", node.Status.Capacity)
+		sriovResourceName, sriovResourceAmount = findSRIOVResource(node)
+		return sriovResourceAmount > 0
+	}, 2*time.Minute, framework.Poll).Should(gomega.BeTrue())
+	framework.Logf("Successfully created device plugin pod, detected %d SRIOV device %q", sriovResourceAmount, sriovResourceName)
+
+	return dpPod, sriovResourceName, sriovResourceAmount
+}
+
+func teardownSRIOVConfigOrFail(f *framework.Framework, dpPod *v1.Pod) {
+	framework.Logf("deleting the SRIOV device plugin pod %s/%s and waiting for container %s removal",
+		dpPod.Namespace, dpPod.Name, dpPod.Spec.Containers[0].Name)
+	deletePodInNamespace(f, dpPod.Namespace, dpPod.Name)
+	waitForContainerRemoval(dpPod.Spec.Containers[0].Name, dpPod.Name, dpPod.Namespace)
```

Comment:
I'm seeing this failure in my setup when I run the test. Looks like 2 min timeout waiting to remove container...

Feb 10 23:39:53.117: INFO: Pod "gu-pod" satisfied condition "Failed"
Feb 10 23:39:53.125: INFO: Waiting for pod gu-pod to disappear
Feb 10 23:39:53.127: INFO: Pod gu-pod no longer exists
Feb 10 23:39:53.127: INFO: deleting the SRIOV device plugin pod kube-system/sriov-device-plugin and waiting for container kube-sriovdp removal

• Failure [498.096 seconds]
[sig-node] Topology Manager [Serial] [Feature:TopologyManager][NodeAlphaFeature:TopologyManager]
_output/local/go/src/k8s.io/kubernetes/test/e2e_node/framework.go:23
  With kubeconfig updated to static CPU Manager policy run the Topology Manager tests
  _output/local/go/src/k8s.io/kubernetes/test/e2e_node/topology_manager_test.go:739
    run Topology Manager node alignment test suite [It]
    _output/local/go/src/k8s.io/kubernetes/test/e2e_node/topology_manager_test.go:695

    Timed out after 120.000s.
    Expected                                                                                                                                                            
        <bool>: false                                                                                                                                                   
    to be true

Reply:
> Can you document the setup required for this test?
> 
> Do we need the actual SR-IOV device plugin or can we have something that is mocked?
> 
> I will let @vpickard finish his review and provide and LGTM before doing a final pass for approval. Thanks for your contribution.

Thanks for your comments! I added the requirements and some documentation here: https://github.com/kubernetes/community/pull/4503

Is there a better place?

Reply:
We investigated on Victor's testing machine, and it seems a local glitch. We'll check again in the near future.

Reply:
Yes, it was a docker issue on my setup. I have it working now.